### DISCUSSION TITLE
Resolve resources in binding conversion

### DISCUSSION THREAD

**Discussion by ProphetLamb** [unoplatform/uno.extensions#2447] [Q&A] [answered]
Good day all,
my desired behavior is that changing the selection in the view model also changes the foreground color of the `TabBarItem`.

My attempt is pretty straight forward, yet I am unable to resolve the theme resource `Theme.Brushes.Primary.Default` to a `Brush` in the convert.
What library function allows this? 

I attempted to work around this using different `Style`s to no avail. Using a style prevents me from inheriting from `Theme.Styles.Buttons.Outline`. Here i have the same issue again: To have my style `BasedOn` the button style I need the `Style` not the resource key for the style, which requires the `TODO_THEME_BRUSH_FROM_KEY` function.

Thanks in advance for any advice.

More generally, I am attempting to setup a `TabBar` for page navigation. 

```csharp
new TabBar().DataContext(new BindableNavigatorModel(), (bar, vm) =>

[...]

new TabBarItem
	.Resources(r => r.Add(icon))
	.Children(
	  new PathIcon()
	    .Data(icon)
	    .Foreground(
	      () => model.SelectedPage,
	      selectedPage => TODO_THEME_BRUSH_FROM_KEY(
	        selectedPage == typeof(MyPage) ? Theme.Brushes.Primary.Default : Theme.Brushes.Primary.Disabled
	      )
	    )
	);
```

```csharp
public partial record NavigatorModel
{
    public NavigatorModel()
    {
        App = (App)Application.Current;
        ArgumentNullException.ThrowIfNull(App.Frame?.CurrentSourcePageType);
        SelectedPage = State.Value(this, () => App.Frame.CurrentSourcePageType);
        App.Frame.Navigated += (_, _) => SelectedPage.Update(_ => App.Frame.CurrentSourcePageType, default);
    }

    public App App { get; }

    public IState<Type> SelectedPage { get; }
}
```

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @ProphetLamb, 

you should be able to use `TryGetValue` and apply the value to the Foreground directly. I tested this with a little bit different sample, so you will need to adjust it for your case. But for the general idea, you can do something like this:

```csharp
.Background(x => x.Binding(() => vm.IsToggleOn).Convert(ConvertIsOnToBrush))
```

```csharp
    private Brush ConvertIsOnToBrush(bool isOn) 
    {
        return isOn ? GetBrush("PrimaryBrush") : GetBrush("TertiaryBrush");
    }

    private Brush GetBrush(string key)
    {
        if (Application.Current.Resources.TryGetValue(key, out var brush))
        {
            return brush as SolidColorBrush;
        }

        throw new KeyNotFoundException($"The resource '{key}' was not found.");
    }
```
